File: code/class_mixin.rb
module Moving
  def walk
    puts "#{self.class.name} is walking"
  end
end

module Interacting
  def talk
    puts "#{self.class.name} is talking"
  end
end

class Human
  include Moving
  include Interacting
end

human = Human.new
human.walk
human.talk

File: code/class_mixin.py
class Moving:
  def walk(self):
    print("%s is walking" % self.__class__.__name__)

class Interacting:
  def talk(self):
    print("%s is talking" % self.__class__.__name__)

class Human(Moving, Interacting):
    pass

human = Human()
human.walk()
human.talk()
